src/misc/support.cpp

```diff
@@ -392,3 +392,25 @@ const std_fs::path &GetExecutablePath()
 	}
 	return exe_path;
 }
+
+const Bit8u DOS_DATE_months[] = {0,  31, 28, 31, 30, 31, 30,
+                                 31, 31, 30, 31, 30, 31};
+
+bool is_date_valid(uint32_t year, uint32_t month, uint32_t day)
+{
+	if (year < 1980 || month > 12 || month == 0 || day == 0)
+		return false;
+	else if (month == 2 && day == 29 &&
+	         (year % 4 != 0 || (year % 100 == 0 && year % 400 != 0)))
+		return false;
+	else if (month != 2 && day > DOS_DATE_months[month])
+		return false;
+	return true;
+}
```

Comment:
Beacuse the if-condition always returns, then the statements after the `if` can be treated as entirely new statements (and not part of the if).  This means we can avoid extra `else` nested levels (which aren't much in this case, but it's a helpful style).

``` c++
if (year < 1980 || month > 12 || month == 0 || day == 0)
	return false;

// February has 28 days except on leap years, when it has 29.
const bool is_leap_year = !(year % 4) && (!(year % 400) || (year % 100));
if (month == 2 && day > (is_leap_year ? 29 : DOS_DATE_months[month]))
	return false;

if (month != 2 && day > DOS_DATE_months[month])
	return false;
return true;
```

https://www.mathworks.com/matlabcentral/answers/469201-write-a-function-called-valid_date-that-takes-three-positive-integer-scalar-inputs-year-month-day

Reply:
Done.